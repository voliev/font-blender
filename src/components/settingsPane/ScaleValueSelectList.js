import React from "react";
import PropTypes from "prop-types";
import { SCALE_VALUES } from "../../constants";
import "./styles/ScaleValueSelectList.css";

const ScaleValueSelectList = ({ disabled, value, scaleChangeHandler }) => (
  <div className="ScaleValueSelectList">
    <label className="label" htmlFor="scale-value">
      Scale value
    </label>
    <select
      disabled={disabled}
      id="scale-value"
      name="scaleValue"
      onChange={scaleChangeHandler}
      value={value}
    >
      {Object.keys(SCALE_VALUES).map(scaleValue => (
        <option key={scaleValue} value={scaleValue}>
          {`${scaleValue} - ${SCALE_VALUES[scaleValue]}`}
        </option>
      ))}
    </select>
  </div>
);

ScaleValueSelectList.propTypes = {
  disabled: PropTypes.bool.isRequired,
  value: PropTypes.number.isRequired,
  scaleChangeHandler: PropTypes.func.isRequired
};

export default React.memo(ScaleValueSelectList);
