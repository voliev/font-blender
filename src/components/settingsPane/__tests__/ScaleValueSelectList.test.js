import React from "react";
import { mount, shallow } from "enzyme";
import ScaleValueSelectList from "../ScaleValueSelectList";

const setup = (renderFn = shallow, propsOverrides = {}) => {
  const props = {
    value: 1.414,
    disabled: false,
    scaleChangeHandler: jest.fn(),
    ...propsOverrides
  };

  return {
    ...props,
    wrapper: renderFn(<ScaleValueSelectList {...props} />)
  };
};

afterEach(() => {
  jest.clearAllMocks();
});

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const { disabled, scaleChangeHandler, value, wrapper } = setup(mount);

  expect(typeof wrapper.prop("value")).toBe("number");
  expect(wrapper.prop("value")).toBe(value);

  expect(typeof wrapper.prop("disabled")).toBe("boolean");
  expect(wrapper.prop("disabled")).toBe(disabled);

  expect(typeof wrapper.prop("scaleChangeHandler")).toBe("function");
  expect(wrapper.prop("scaleChangeHandler")).toBe(scaleChangeHandler);
});

it("should handle select change", () => {
  const { scaleChangeHandler, wrapper } = setup();

  const event = {
    target: {
      value: 1.2
    }
  };

  wrapper.find("select").simulate("change", event);

  expect(scaleChangeHandler).toHaveBeenCalledTimes(1);
  expect(scaleChangeHandler).toHaveBeenCalledWith(event);
});
