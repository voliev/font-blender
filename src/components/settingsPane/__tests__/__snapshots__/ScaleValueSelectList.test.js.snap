// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="ScaleValueSelectList"
>
  <label
    className="label"
    htmlFor="scale-value"
  >
    Scale value
  </label>
  <select
    disabled={false}
    id="scale-value"
    name="scaleValue"
    onChange={[MockFunction]}
    value={1.414}
  >
    <option
      key="1.067"
      value="1.067"
    >
      1.067 - Minor Second
    </option>
    <option
      key="1.125"
      value="1.125"
    >
      1.125 - Major Second
    </option>
    <option
      key="1.2"
      value="1.2"
    >
      1.2 - Minor Third
    </option>
    <option
      key="1.25"
      value="1.25"
    >
      1.25 - Major Third
    </option>
    <option
      key="1.333"
      value="1.333"
    >
      1.333 - Perfect Fourth
    </option>
    <option
      key="1.414"
      value="1.414"
    >
      1.414 - Augmented Fourth
    </option>
    <option
      key="1.5"
      value="1.5"
    >
      1.5 - Perfect Fifth
    </option>
    <option
      key="1.618"
      value="1.618"
    >
      1.618 - Golden Ratio
    </option>
  </select>
</div>
`;
